Task10_Auth/src/test/java/CreateUsersTests.java

```diff
@@ -0,0 +1,108 @@
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.jupiter.api.Test;
+import reseponsesDTO.UsersDto;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static reseponsesDTO.UsersDto.randomSex;
+
+public class CreateUsersTests {
+    private ZipCodesClient zipCodesClient = new ZipCodesClient();
+    private UsersClient usersClient = new UsersClient();
+    private static final Random random = new Random();
+
+    @Test
+    public void createUserTest() {
+        ResponseWrapper getZipCodesResponseBeforeCreation = zipCodesClient.getZipCodes();
+        int age = RandomUtils.nextInt(1, 99);
+        String name = RandomStringUtils.random(5, true, false);
+        UsersDto.sex userSex = randomSex();
+        int zipCodesSize = getZipCodesResponseBeforeCreation.getResponseBodyZipCodes().size();
+        String exactZipCode = getZipCodesResponseBeforeCreation.getResponseBodyZipCodes().get(random.nextInt(zipCodesSize));
+        int exactZipCodesCountBeforeCreation = Collections.frequency(getZipCodesResponseBeforeCreation.getResponseBodyZipCodes(), exactZipCode);
+
+        UsersDto userToAdd = new UsersDto();
+        userToAdd = userToAdd.createUserDtoObject(age, name, userSex, exactZipCode);
+
+        ResponseWrapper createUserResponse = usersClient.createUser(userToAdd);
+
+
+        ResponseWrapper getZipCodesResponseAfterCreation = zipCodesClient.getZipCodes();
+        int exactZipCodesCountAfterCreation = Collections.frequency(getZipCodesResponseAfterCreation.getResponseBodyZipCodes(), exactZipCode);
+        UsersDto addedUser = new UsersDto();
+        ResponseWrapper getUsersResponse = usersClient.getUsers();
+        ArrayList<UsersDto> usersList = getUsersResponse.getResponseBodyUsers();
+        if (usersList.contains(userToAdd)) {
+            addedUser = usersList.get(usersList.indexOf(userToAdd));
+        }
```

Comment:
I think that is extra step and you can use userToAdd as expected result. Your contains operation must be part of assertion.

Reply:
done